miner.go

```diff
@@ -139,6 +145,10 @@ func NewMiner(ctx context.Context) (*Miner, error) {
 		return nil, fmt.Errorf("no devices started")
 	}
 
+	if cfg.Benchmark {
+		return newBenchmarkMiner(devices)
```

Comment:
Returning early here means that `m.workDone` and `m.started` are not being set. 

- `workDone` is only used when submitting work (which doesnt happen in benchmark mode).
- The only impact of not setting `started` afaict is that the uptime reported by the status endpoint will be incorrect. 

Overall these probably don't matter, but because they can be so easily fixed we might as well do it to prevent any issues cropping up in future.

Reply:
Good point. I've rewritten to make the init code more uniform.